store/sqlstore/post_store.go

```diff
@@ -661,6 +662,48 @@ func (s *SqlPostStore) getPostsAround(channelId string, postId string, limit int
 	return list, nil
 }
 
+func (s *SqlPostStore) GetPostBeforeTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	return s.getPostAroundTime(channelId, time, true)
+}
+
+func (s *SqlPostStore) GetPostAfterTime(channelId string, time int64) (*model.Post, *model.AppError) {
+	return s.getPostAroundTime(channelId, time, false)
+}
+
+func (s *SqlPostStore) getPostAroundTime(channelId string, time int64, before bool) (*model.Post, *model.AppError) {
+	var direction string
+	var sort string
+	if before {
```

Comment:
Might we leverage squirrel to build the query instead?

Reply:
Done.  Also did the same with `GetPostAfterTime`